```
johnrtitor@darwin01 ~ % nix build "github:nixos/nixpkgs/refs/pull/394677/merge#kitty" --print-out-paths        
error: builder for '/nix/store/xgbxng4w18k7gps7clpyzrj3i0fwk5fd-kitty-0.41.0.drv' failed with exit code 2;
       last 25 log lines:
       > ======================================================================
       > FAIL: test_fish_integration (kitty_tests.shell_integration.ShellIntegrationWithKitten.test_fish_integration)
       > ----------------------------------------------------------------------
       > Traceback (most recent call last):
       >   File "/private/tmp/nix-build-kitty-0.41.0.drv-0/source/kitty/launcher/kitty.app/Contents/MacOS/../../../../../kitty_tests/shell_integration.py", line 200, in test_fish_integration
       >     self.ae(pty.screen_contents(), q)
       >     completions_dir = '/private/tmp/nix-build-kitty-0.41.0.drv-0/source/shell-integration/fish/vendor_completions.d'
       >     fish_prompt = 'left>'
       >     pty = <kitty_tests.PTY object at 0x347d47e00>
       >     q = 'left>                                                                                                                                                                         <right'
       >     right_prompt = '<right'
       >     self = <kitty_tests.shell_integration.ShellIntegrationWithKitten testMethod=test_fish_integration>
       > AssertionError: 'warning: notify_register_file_descriptor([265 chars]ight' != 'left>                                    [135 chars]ight'
       > - warning: notify_register_file_descriptor() failed with status 9.
       > - warning: Universal variable notifications may not be received.
       >   left>                                                                                                                                                                         <right
       >
       >
       > ----------------------------------------------------------------------
       > Ran 152 tests in 23.263s
       >
       > FAILED (failures=2, errors=1, skipped=4)
       > All Go tests succeeded, ran in 23.7 seconds
       > Error: Some tests failed!
       > make: *** [Makefile:16: test] Error 1
       For full logs, run 'nix log /nix/store/xgbxng4w18k7gps7clpyzrj3i0fwk5fd-kitty-0.41.0.drv'.
nix build "github:nixos/nixpkgs/refs/pull/394677/merge#kitty"   4.72s user 17.14s system 18% cpu 2:01.21 total
```

Tested on Nix Community darwin builder.

--- a/pkgs/by-name/ki/kitty/package.nix
+++ b/pkgs/by-name/ki/kitty/package.nix
@@ -231,9 +231,16 @@ buildPythonApplication rec {
 
   # skip failing tests due to darwin sandbox
   preCheck = lib.optionalString stdenv.hostPlatform.isDarwin ''
-    # can be re-enabled with the next kitty release, see https://github.com/kovidgoyal/kitty/pull/7939
+
     substituteInPlace kitty_tests/file_transmission.py \
       --replace test_transfer_send dont_test_transfer_send
+
+    substituteInPlace kitty_tests/ssh.py \
+      --replace test_ssh_connection_data no_test_ssh_connection_data \
+
+    substituteInPlace kitty_tests/shell_integration.py \
+      --replace test_fish_integration no_test_fish_integration
+
     # theme collection test starts an http server
     rm tools/themes/collection_test.go
     # passwd_test tries to exec /usr/bin/dscl

--- a/pkgs/by-name/ki/kitty/package.nix
+++ b/pkgs/by-name/ki/kitty/package.nix
@@ -44,8 +44,8 @@
   zsh,
   fish,
   nixosTests,
-  go_1_23,
-  buildGo123Module,
+  go_1_24,
+  buildGo124Module,
   nix-update-script,
   makeBinaryWrapper,
   autoSignDarwinBinariesHook,
@@ -56,21 +56,21 @@
 with python3Packages;
 buildPythonApplication rec {
   pname = "kitty";
-  version = "0.40.1";
+  version = "0.41.1";
   format = "other";
 
   src = fetchFromGitHub {
     owner = "kovidgoyal";
     repo = "kitty";
     tag = "v${version}";
-    hash = "sha256-ta9MTbSkIkowkd5zBUbtauFNGVRDgxof1SSQXuEgFTQ=";
+    hash = "sha256-oTkzFEPgbFa2wPBJxh/9ZbK8liM9isWGEwExJq5/h2o=";
   };
 
   goModules =
-    (buildGo123Module {
+    (buildGo124Module {
       pname = "kitty-go-modules";
       inherit src version;
-      vendorHash = "sha256-wr5R2X+lV8vVVWsDYLLSaz7HRNOB7Zzk/a7knsdDlXs=";
+      vendorHash = "sha256-ld3cGJUjoi3od6gINyGE7fQodl9CSKmakJ1CPLMX+Ss=";
     }).goModules;
 
   buildInputs =
@@ -124,7 +124,7 @@ buildPythonApplication rec {
       sphinx-copybutton
       sphinxext-opengraph
       sphinx-inline-tabs
-      go_1_23
+      go_1_24
       fontconfig
       makeBinaryWrapper
     ]

--- a/pkgs/by-name/ki/kitty/package.nix
+++ b/pkgs/by-name/ki/kitty/package.nix
@@ -241,6 +241,9 @@ buildPythonApplication rec {
     substituteInPlace kitty_tests/shell_integration.py \
       --replace test_fish_integration no_test_fish_integration
 
+    substituteInPlace kitty_tests/fonts.py \
+      --replace test_fallback_font_not_last_resort no_test_fallback_font_not_last_resort
+
     # theme collection test starts an http server
     rm tools/themes/collection_test.go
     # passwd_test tries to exec /usr/bin/dscl

